app/webpacker/components/RegistrationsV2/Register/CompetingStep.jsx

```diff
@@ -237,29 +235,41 @@ export default function CompetingStep({
           </Message>
         )}
 
-        <Form>
-          <Form.Field>
+        <Form onSubmit={(event) => {
+          event.preventDefault();
+          if (shouldShowUpdateButton) {
+            attemptAction(actionUpdateRegistration, { checkForChanges: true });
+          } else if (shouldShowReRegisterButton) {
+            attemptAction(actionReRegister);
+          } else {
+            attemptAction(actionCreateRegistration);
+          }
+        }}
+        >
```

Comment:
Move this into its own callback `const` (and potentially even wrap in `useCallback`? Although that's optional if you feel that it doesn't make sense)